### PR TITLE
Patch collection_name and collection_namespace to inventory

### PR DESCRIPTION
**Pull request by rohitthakur2590**
Signed-off-by: Rohit Thakur <rohitthakur2590@outlook.com>
This will be used by ansible-network-ee-integration-tests jobs.

### CODE REVIEW

**Comment by pabelanger** [comment]
The better way we should do this, is expose collection_name / collection_namespace to the job variables, and allow us to write them into the inventory file. Given we likely will have multiple appliances wanting to test against ansible.network.

You can propose this change today, but using the existing vyos test job but try to get it to run ansible.network collection testing.  https://github.com/ansible/network-ee/blob/main/playbooks/network-ee-integration-tests/pre.yaml#L31-L69 is how we modify the inventory file during testing. We should do the same for collection_name / collection_namespace

File: tests/integration/inventories/vyos.yaml [hunk 1]
```diff
@@ -15,5 +15,5 @@ appliance:
     # NOTE(pabelanger): With stable-2.9 / stable-2.11 we need to force
     # /bin/python3.8 and not discover /usr/libexec/platform-python (python3.6)
     # ansible_python_interpreter: /bin/python3.8
-    collection_name: vyos
-    collection_namespace: vyos
+    collection_name: "{{ collection_name }}"
+    collection_namespace: "{{ collection_namespace }}"
```
Comment (pabelanger) [inline tests/integration/inventories/vyos.yaml:19]: we don't want to change this file, we should update the playbook I linked before. Where we overwrite this file with the zuul job.vars if defined.

**Comment by rohitthakur2590** [comment]
recheck

File: playbooks/network-ee-integration-tests/pre.yaml [hunk 2]
```diff
@@ -68,6 +68,38 @@
...
+      set_fact:
+        _inventory_data: "{{ _inventory_data | combine(newdata, recursive=True) }}"
+      vars:
+        newdata:
+          appliance:
+            vars:
+              collection_name: "{{ collection_name }}"
+      when: collection_name is defined
+
+    - name: Write appliance inventory file with updated collection_name
```
Comment (pabelanger) [inline playbooks/network-ee-integration-tests/pre.yaml:81]: we don't need to write this file multiple times, just on line 66

File: playbooks/network-ee-integration-tests/pre.yaml [hunk 3]
```diff
@@ -68,6 +68,38 @@
...
+          appliance:
+            vars:
+              collection_namespace: "{{ collection_namespace }}"
+      when: collection_namespace is defined
+
+    - name: Write appliance inventory file with updated collection namespace
+      copy:
+        content: "{{ _inventory_data | to_nice_yaml(indent=2, explicit_start=True) }}"
+        dest: "{{ zuul.projects['github.com/ansible/network-ee'].src_dir }}/tests/integration/inventories/{{ _network_os }}.yaml"
+      when: collection_namespace is defined
```
Comment (pabelanger) [inline playbooks/network-ee-integration-tests/pre.yaml:101]: same as above

File: playbooks/network-ee-integration-tests/pre.yaml [hunk 4]
```diff
@@ -68,6 +68,38 @@
...
 
+    - name: Patch collection_name inventory data
+      set_fact:
+        _inventory_data: "{{ _inventory_data | combine(newdata, recursive=True) }}"
+      vars:
+        newdata:
+          appliance:
+            vars:
+              collection_name: "{{ collection_name }}"
+      when: collection_name is defined
```
Comment (pabelanger) [inline playbooks/network-ee-integration-tests/pre.yaml:79]: this should move above line 66

File: playbooks/network-ee-integration-tests/pre.yaml [hunk 5]
```diff
@@ -68,6 +68,38 @@
...
+
+    - name: Patch collection_namespace inventory data
+      set_fact:
+        _inventory_data: "{{ _inventory_data | combine(newdata, recursive=True) }}"
+      vars:
+        newdata:
+          appliance:
+            vars:
+              collection_namespace: "{{ collection_namespace }}"
+      when: collection_namespace is defined
```
Comment (pabelanger) [inline playbooks/network-ee-integration-tests/pre.yaml:95]: same above line 66

**Comment by rohitthakur2590** [comment]
@pabelanger  Can we merged this now, as VyOS Jobs  (https://github.com/ansible-collections/vyos.vyos/pull/207 ) didn't break.

**Comment by pabelanger** [comment]
thanks! You should able be able to atleast parent to the vyos integration job now, setup ansible.network as your collection name, and test with vyos.  until we land cisco.ios jobs

**Pull request merged**